sql/core/src/test/scala/org/apache/spark/sql/ApproximatePercentileQuerySuite.scala

```diff
@@ -200,9 +200,9 @@ class ApproximatePercentileQuerySuite extends QueryTest with SharedSparkSession
       checkAnswer(
         spark.sql(
           s"""SELECT
-              |  percentile_approx(null, 0.5),
+              |  percentile_approx(cast(null as int), 0.5),
```

Comment:
I agree that null literal should not be supported, but it's really trivial to support NullType in type coercions. Shall we add the support and avoid this breaking change?